tokio/src/sync/mpsc/chan.rs

```diff
@@ -263,6 +267,26 @@ impl<T, S: Semaphore> Rx<T, S> {
             }
         })
     }
+
+    /// Receives the next value without blocking
+    pub(crate) fn try_recv(&mut self) -> Result<T, TryRecvError> {
+        use super::block::Read::*;
+
+        self.inner.rx_fields.with_mut(|rx_fields_ptr| {
+            let rx_fields = unsafe { &mut *rx_fields_ptr };
+            loop {
+                match rx_fields.list.pop(&self.inner.tx) {
+                    Some(Value(value)) => {
+                        self.inner.semaphore.add_permit();
+                        return Ok(value);
+                    }
+                    Some(Closed) => return Err(TryRecvError::Closed),
+                    Some(NotReady) => thread::yield_now(),
```

Comment:
Should this be using [`std::hint::spin_loop`][1]?

[1]: https://doc.rust-lang.org/stable/std/hint/fn.spin_loop.html

Reply:
Good question. I borrowed it from https://github.com/tokio-rs/tokio/blob/master/tokio/src/sync/mpsc/block.rs#L348. It has a comment that convinced me this is the right thing to use. But is this comment actually correct?  Unclear. Any thoughts ?

Reply:
Also, what about this bit of docs: https://docs.rs/loom/0.4.0/loom/#yielding

Reply:
The comment you linked appears to be wrong, as [the non-loom](https://github.com/tokio-rs/tokio/blob/master/tokio/src/loom/std/mod.rs) version just re-exports `std::thread`. It also has a comment on MSRV.